jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -83,58 +84,16 @@ static JarType determineJarType(Path jarPath) throws IOException {
    * @throws IOException if I/O error occurs when opening the jar file or if temporary directory
    *     provided doesn't exist
    */
-  static List<FileEntriesLayer> createExplodedModeLayersForStandardJar(
-      Path jarPath, Path tempDirPath) throws IOException, IllegalArgumentException {
-    Path localExplodedJarRoot = tempDirPath;
-    ZipUtil.unzip(jarPath, localExplodedJarRoot);
-    List<FileEntriesLayer> layers = new ArrayList<>();
-
-    // Get dependencies from Class-Path in the jar's manifest and add a layer each for non-snapshot
-    // and snapshot dependencies. If Class-Path is not present in the jar's manifest then skip
-    // adding the dependencies layers.
-    String classPath = null;
-    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
-      classPath = jarFile.getManifest().getMainAttributes().getValue(Attributes.Name.CLASS_PATH);
-    }
-    if (classPath != null) {
-      Predicate<String> isSnapshot = name -> name.contains("SNAPSHOT");
-      List<String> allDependencies = Splitter.onPattern("\\s+").splitToList(classPath.trim());
-      List<Path> nonSnapshotDependencies =
-          allDependencies
-              .stream()
-              .filter(isSnapshot.negate())
-              .map(Paths::get)
-              .collect(Collectors.toList());
-      List<Path> snapshotDependencies =
-          allDependencies.stream().filter(isSnapshot).map(Paths::get).collect(Collectors.toList());
-      Path jarParent = jarPath.getParent() == null ? Paths.get("") : jarPath.getParent();
-      if (!nonSnapshotDependencies.isEmpty()) {
-        FileEntriesLayer.Builder nonSnapshotDependenciesLayerBuilder =
-            FileEntriesLayer.builder().setName(DEPENDENCIES);
-        nonSnapshotDependencies.forEach(
-            path ->
-                addDependency(
-                    nonSnapshotDependenciesLayerBuilder,
-                    jarParent.resolve(path),
-                    APP_ROOT.resolve("dependencies").resolve(path.getFileName())));
-        layers.add(nonSnapshotDependenciesLayerBuilder.build());
-      }
-      if (!snapshotDependencies.isEmpty()) {
-        FileEntriesLayer.Builder snapshotDependenciesLayerBuilder =
-            FileEntriesLayer.builder().setName(SNAPSHOT_DEPENDENCIES);
-        snapshotDependencies.forEach(
-            path ->
-                addDependency(
-                    snapshotDependenciesLayerBuilder,
-                    jarParent.resolve(path),
-                    APP_ROOT.resolve("dependencies").resolve(path.getFileName())));
-        layers.add(snapshotDependenciesLayerBuilder.build());
-      }
-    }
+  static List<FileEntriesLayer> createLayersForExplodedStandard(Path jarPath, Path tempDirPath)
+      throws IOException {
+    // Add dependencies layers.
+    List<FileEntriesLayer> layers = getDependenciesLayers(jarPath, "exploded");
```

Comment:
We have the enum class defined in this Java package (`c.g.c.t.jib.cli.jar.ProcessingMode`). Why not use it?

Reply:
Oh yes, good idea!